packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -198,6 +201,15 @@ export interface FunctionProps {
    * You can also add event sources using `addEventSource`.
    */
   events?: IEventSource[];
+
+  /**
+   * The number of days log events are kept in CloudWatch Logs. When updating
+   * this property, unsetting it doesn't remove the log retention policy. To
```

Comment:
Wouldn't it make more sense for the unset behavior to update the retention to `Infinity` explicitly?

Reply:
We have two implementation choices here:
- The custom resource is added *only* when `logRetentionDays` is defined. In this case, unsetting `logRetentionDays` will remove the custom resource and since we cannot discriminate between this and a delete coming from a function delete or stack delete we have to find a way to go back to 'never expire'. This is the current implementation.
- Always create the custom resource (even if `logRetentionDays` is `undefined` for all Lambda functions in stack. In this case, we can discriminate easily between unsetting `logRetentionDays` and a real delete.

Preferences?

For option 2, units tests that match the whole stack and integ tests have to be updated.

Reply:
I see the issue. In that case, I think option 1 makes more sense. It doesn't make sense that a Lambda without log retention configuration will add a custom resource to each stack...